### PR TITLE
feature / versioning

### PR DESCRIPTION
**Pull request by pvilalta**
Hello team,

This PR brings the versioning feature

Paul

### CODE REVIEW

**Comment by amineau** [comment]
https://platform-dev-versioning.bimdata.io/

File: src/components/specific/versioning/versioning-main/VersioningMain.vue [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        <div class="versioning-main__content__empty-area">
+          <div class="versioning-main__content__empty-area__info">
+            <img src="/static/versioning/versioning-empty-history.svg" />
+            <span>{{ $t("Versioning.emptyArea") }}</span>
+          </div>
+        </div>
+      </template>
+      <template v-else>
+        <div class="versioning-main__content__list">
+          <template v-for="(doc, index) of allDocVersions" :key="index">
```
Comment (NicolasRichel) [inline src/components/specific/versioning/versioning-main/VersioningMain.vue]: I see an **index** as `key` here.
Did you check that it does not result in the same bug that we discuss last time ?
(As a reminder: the UI was not properly updated when deleting the first element due to the way Vue uses `key` attribute internally to identify nodes).

**Review by NicolasRichel — CHANGES_REQUESTED** [review]

**Review by GabBimdata — APPROVED** [review]

File: src/components/specific/files/files-manager/FilesManager.vue [hunk 2]
```diff
@@ -372,6 +388,24 @@ export default {
...
+        showVersioningManager.value = true;
+        showAccessManager.value = false;
+        showVisaManager.value = false;
+        showSidePanel.value = true;
+      }
+    };
+
+    const closeVersioningManager = () => {
+      showSidePanel.value = false;
+      setTimeout(() => {
```
Comment (Kurtil) [inline src/components/specific/files/files-manager/FilesManager.vue:402]: Why is there a setTimout here ? Can a $nextTick work instead ?

**Review by Kurtil — CHANGES_REQUESTED** [review]

File: src/components/specific/files/files-manager/FilesManager.vue (hunk 2 above)
Comment (pvilalta) [reply src/components/specific/files/files-manager/FilesManager.vue:402]: nextTick work as well, but for a smoother transition setTimeout is better

**Review by Kurtil — APPROVED** [review]

**Review by NicolasRichel — APPROVED** [review]

**Pull request merged**